src/main.cc

```diff
@@ -7,208 +7,131 @@
-
+#include <QtCore/QCommandLineParser>
 #ifdef Q_OS_MACOS
-    #include <QtCore/QProcessEnvironment>
+#include <QtCore/QProcessEnvironment>
 #endif
+#include <QtQuick/QQuickWindow>
+#include <QtWidgets/QApplication>
 
 #include "QGCApplication.h"
+#include "QGCCommandLineParser.h"
 #include "QGCLogging.h"
-#include "CmdLineOptParser.h"
+#include "UnitTestList.h"
 
 #if !defined(Q_OS_ANDROID) && !defined(Q_OS_IOS)
-    #include <QtWidgets/QMessageBox>
-    #include "RunGuard.h"
+#include <QtWidgets/QMessageBox>
+#include <QtWidgets/QErrorMessage>
+#include "RunGuard.h"
 #endif
 
 #ifdef Q_OS_ANDROID
-    #include "AndroidInterface.h"
-#endif
-
-#ifdef Q_OS_LINUX
-#ifndef Q_OS_ANDROID
-    #include "SignalHandler.h"
-#endif
-#endif
-
-#ifdef QT_DEBUG
-#ifdef QGC_UNITTEST_BUILD
-    #include "UnitTestList.h"
+#include "AndroidInterface.h"
+#elif defined(Q_OS_LINUX)
+#include <unistd.h>
+#include "SignalHandler.h"
 #endif
 
-#ifdef Q_OS_WIN
-
-#include <crtdbg.h>
-#include <windows.h>
-#include <iostream>
-
-/// @brief CRT Report Hook installed using _CrtSetReportHook. We install this hook when
-/// we don't want asserts to pop a dialog on windows.
-int WindowsCrtReportHook(int reportType, char* message, int* returnValue)
+int main(int argc, char *argv[])
 {
-    Q_UNUSED(reportType);
-
-    std::cerr << message << std::endl;  // Output message to stderr
-    *returnValue = 0;                   // Don't break into debugger
-    return true;                        // We handled this fully ourselves
-}
-
-#endif // Q_OS_WIN
-
-#endif // QT_DEBUG
+    QStringList rawArgs;
+    for (int i = 1; i < argc; ++i) {
+        rawArgs << QString::fromLocal8Bit(argv[i]);
+    }
 
-//-----------------------------------------------------------------------------
-/**
- * @brief Starts the application
- *
- * @param argc Number of commandline arguments
- * @param argv Commandline arguments
- * @return exit code, 0 for normal exit and !=0 for error cases
- */
+    QCommandLineParser preappParser;
+    preappParser.setSingleDashWordOptionMode(QCommandLineParser::ParseAsLongOptions);
+    (void) preappParser.addOption({QStringLiteral("desktop"), QStringLiteral("Use desktop OpenGL")});
+    (void) preappParser.addOption({QStringLiteral("software"), QStringLiteral("Use software OpenGL")});
+    (void) preappParser.addOption({QStringLiteral("gles"), QStringLiteral("Use OpenGLES")});
+    (void) preappParser.addOption({QStringLiteral("shared"), QStringLiteral("Share OpenGL Contexts")});
+    (void) preappParser.parse(rawArgs);
+
+    if (preappParser.isSet("desktop")) {
+        QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);
+    } else if (preappParser.isSet("software")) {
+        QCoreApplication::setAttribute(Qt::AA_UseSoftwareOpenGL);
+    } else if (preappParser.isSet("gles")) {
+        QCoreApplication::setAttribute(Qt::AA_UseOpenGLES);
+    } else if (preappParser.isSet("shared")) {
+        QCoreApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
```

Comment:
@DonLakeFlyer I'm pretty sure this option resolves the issue mentioned at the top of https://gstreamer.freedesktop.org/documentation/qml6/qml6glsink.html?gi-language=c#qml6glsink